internal/provider/resource_application_identity.go

```diff
@@ -79,11 +79,11 @@ var azureApplicationIdentityOutputs = tfsdk.Attribute{
 	Computed:    true,
 	Description: "Azure application and service principal configuration",
 	Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
-		"application_id": {
+		"service_principal_object_id": {
```

Comment:
Maybe we keep `object` in golang vars but drop it here for a simpler interface.

Reply:
Yeah, when I named these I tried to keep them similar to the counterparts I identified in the official `azurerm` provider. The idea being that it will be a similar interface for anybody that already uses azure for managing terraform.

Reply:
The discovery we made yesterday though is the `Application.go` package in the Hamilton repo is using **Azure Enterprise Applications**, which is different than **Azure App Registration** (which is where service principals are configured). In some edge cases, the appID of the service principal is not the appID of the enterprise application. I made the change to add clarity of what the actual value is coming through as, because the `service_principal_object_id` is for API requests, but the service principal's appID is used for auth/roles/etc.

Basically, we were slamming our heads against the wall yesterday trying to figure out why the appID of the application, which should be giving us the ability to auth to the ACR, was failing.

Reply:
Since this is the "exposed" interface that we'd interact w/ in Terraform. Let's make this name `service_principal_id`

Reply:
If this is the exposed object we interact with, I think it would be best to make it clear about what exactly is being passed since `id` is vague in Azure's terms.

Reply:
Personally, I agree since Azure is hard! But Chris has a good point here. In Terraform for Azure, I haven't seen `service_principal_object_id` but I have seen `service_principal_id`. He's saying that we should match what's used in the Azure provider and other Azure TF so when people see this value, they know what it means, even if object id is technically clearer.

Reply:
Change has been made